### PR TITLE
p192: add `ecdsa` feature

### PR DESCRIPTION
**Pull request by halvors**
Add ECDSA support for P-192, tests not implemented yet due to lacking signer support in P-192 and also requirement to use `PrehashSigner::sign_prehash()` since `ecdsa` crate currently only supports sha2 hashing algorithms for signing

### CODE REVIEW

File: p192/src/ecdsa.rs [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+/// ECDSA/P-192 signing key
+#[cfg(feature = "ecdsa")]
+pub type SigningKey = ecdsa_core::SigningKey<NistP192>;
+
+/// ECDSA/P-192 verification key (i.e. public key)
+#[cfg(feature = "ecdsa")]
+pub type VerifyingKey = ecdsa_core::VerifyingKey<NistP192>;
+
+#[cfg(feature = "ecdsa")]
+impl SignPrimitive<NistP192> for Scalar {}
```
Comment (tarcieri) [inline p192/src/ecdsa.rs:64]: I'd prefer to leave the signing components out for now, per NIST's guidance

Comment (halvors) [reply p192/src/ecdsa.rs]: Done

**Comment by halvors** [comment]
@tarcieri I have added test vectors as well test pass (only verifying), i am a little bit unsure about where to obtain `wycheproof` but it seem to require signing, is only verifying test with the test vectors sufficient for P-192?

File: p192/src/ecdsa.rs [hunk 2]
```diff
@@ -13,29 +13,6 @@
 //! - `ecdsa`: provides `ecdsa-core` features plus the [`SigningKey`] and
 //!   [`VerifyingKey`] types which natively implement ECDSA/P-192 signing and
 //!   verification.
-//!
```
Comment (halvors) [inline p192/src/ecdsa.rs]: What to do about this, impossible to obtain the signing key without signing support, should we remove this or use a static set example key?

File: p192/src/ecdsa.rs [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+/// ECDSA/P-192 signature (fixed-size)
+pub type Signature = ecdsa_core::Signature<NistP192>;
+
+/// ECDSA/P-192 signature (ASN.1 DER encoded)
+pub type DerSignature = ecdsa_core::der::Signature<NistP192>;
+
+/// ECDSA/P-192 signing key
+#[cfg(feature = "ecdsa")]
+pub type SigningKey = ecdsa_core::SigningKey<NistP192>;
```
Comment (tarcieri) [inline p192/src/ecdsa.rs:31]: Can you remove this (and other mentions of `SigningKey`) as well?

It won't work without `SignPrimitive`.

File: p192/src/ecdsa.rs (hunk 2 above)
Comment (tarcieri) [reply p192/src/ecdsa.rs]: Remove it from the docs

**Comment by tarcieri** [comment]
Correct you can't run the Wycheproof tests without a `SigningKey`, so just include the NIST CAVS vectors

File: p192/src/ecdsa.rs (hunk 3 above)
Comment (halvors) [reply p192/src/ecdsa.rs]: Sure, done

**Comment by halvors** [comment]
> Needs rustfmt but otherwise looks good.
> 
> Edit: I'll take care of rustfmt.

Thanks, I am sorry that I forgot

**Pull request merged**